src/_P037_MQTTImport.ino

```diff
@@ -53,6 +59,75 @@ String P037_getMQTTLastTopicPart(const String& topic) {
 
 # endif // if P037_MAPPING_SUPPORT || P037_JSON_SUPPORT
 
+bool P037_addEventToQueue(struct EventStruct *event, String& newEvent) {
+  if (newEvent.isEmpty()) { return false; }
+  bool result    = true;
+  uint8_t reason = 0;
+
+  if (result && P037_DEDUPLICATE_EVENTS) { // Check if event is already queued
+    String queuedEvent;
+    reason = 1;                            // Deduping started, so that might be the reason
+    # if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+    String log;
+
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+      addLog(LOG_LEVEL_DEBUG, F("P037: addEvent deduping"));
+      addLog(LOG_LEVEL_DEBUG, newEvent);
+      log.reserve(64);
+    }
+    # endif // if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+    for (auto iter = eventQueue.begin(); iter != eventQueue.end() && result; ++iter) {
+      queuedEvent = String(*iter);
+      # if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+      if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+        log += queuedEvent;
+        log += ';';
+      }
+      # endif // if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+      result = !queuedEvent.equalsIgnoreCase(newEvent); // IgnoreCase-compare!
```

Comment:
Hmm too bad you can't use `find` here on the queue, simply because you need to match case insensitive.
But I guess it is good to have such a dedup-option for events anyway, so why not add this dedup option to the add function of the eventQueue? (default parameter for dedup set to false, but we may later make it a setting)

Reply:
Hm, maybe I can use `find`, as the events will always be the same, because they come from an external source, and often device-generated.

Reply:
Switched to case-sensitive comparison. Now use `find` for matching events.